specification/provisioningservices/resource-manager/Microsoft.Devices/2017-08-21-preview/iotdps.json

```diff
@@ -0,0 +1,1600 @@
+					"type": "array",
+					"items": {
+						"$ref": "#/definitions/SharedAccessSignatureAuthorizationRule"
+					}
+				},
+				"ipFilterRules": {
+					"type": "array",
+					"items": {
+						"$ref": "#/definitions/IpFilterRule"
+					}
+				},
+				"hostName": {
+					"type": "string"
+				},
+				"eventHubEndpoints": {
+					"type": "object",
+					"additionalProperties": {
+						"$ref": "#/definitions/EventHubProperties"
+					}
+				},
+				"routing": {
+					"$ref": "#/definitions/RoutingProperties"
+				},
+				"storageEndpoints": {
+					"type": "object",
+					"additionalProperties": {
+						"$ref": "#/definitions/StorageEndpointProperties"
+					}
+				},
+				"messagingEndpoints": {
+					"type": "object",
+					"additionalProperties": {
+						"$ref": "#/definitions/MessagingEndpointProperties"
+					}
+				},
+				"enableFileUploadNotifications": {
+					"type": "boolean"
+				},
+				"cloudToDevice": {
+					"$ref": "#/definitions/CloudToDeviceProperties"
+				},
+				"comments": {
+					"type": "string"
+				},
+				"systemProperties": {
+					"type": "object",
+					"additionalProperties": {
+						"type": "string"
+					}
+				},
+				"features": {
+					"enum": ["None", "DeviceManagement"],
+					"type": "string",
+					"x-ms-enum": {
+						"name": "Capabilities",
+						"modelAsString": "True"
+					}
+				}
+			}
+		},
+		"CertificateResponse": {
+			"type": "object",
+			"properties": {
+				"name": {
+					"type": "string"
+				},
+				"etag": {
+					"type": "string"
+				},
+				"subject": {
+					"type": "string"
+				},
+				"expiry": {
+					"format": "date-time",
+					"type": "string"
+				},
+				"thumbprint": {
+					"type": "string"
+				},
+				"isVerified": {
+					"type": "boolean"
+				},
+				"created": {
+					"format": "date-time",
+					"type": "string"
+				},
+				"updated": {
+					"format": "date-time",
+					"type": "string"
+				}
+			}
+		},
+		"IotDpsSkuInfo": {
+			"type": "object",
+			"properties": {
+				"name": {
+					"enum": ["S1"],
+					"type": "string",
+					"x-ms-enum": {
+						"name": "IotDpsSku",
+						"modelAsString": "True"
+					}
+				},
+				"tier": {
+					"type": "string",
+					"readOnly": true
+				},
+				"capacity": {
+					"format": "int64",
+					"type": "integer"
+				}
+			}
+		},
+		"ProvisioningServiceDescription": {
+			"type": "object",
+			"properties": {
+				"etag": {
+					"type": "string"
+				},
+				"name": {
+					"type": "string",
+					"readOnly": "true"
+				},
+				"location": {
+					"type": "string"
+				},
+				"properties": {
+					"$ref": "#/definitions/IotDpsPropertiesDescription"
+				},
+				"type": {
+					"type": "string",
+					"readOnly": true
+				},
+				"id": {
+					"type": "string",
+					"readOnly": true
+				},
+				"tags": {
+					"type": "object",
+					"additionalProperties": {
+						"type": "string"
+					}
+				}
+			}
+		},
+		"IotDpsPropertiesDescription": {
+			"type": "object",
+			"properties": {
+				"state": {
```

Comment:
Will these states come into effect after resource has been provisioned? Or will the resource transition between these states during provisioning? if yes, we may just need the provisioningState property